spec/controllers/admin/users_controller_spec.rb

```diff
@@ -2,89 +2,115 @@
 
 RSpec.describe Admin::UsersController, type: :controller do
   before do
+    allow(controller).to receive(:authorize) { authorized? }
+    allow(controller).to receive(:pundit_policy_authorized?) { authorized? }
+
     sign_in(user)
   end
 
-  let(:user) { build_stubbed(:user) }
+  let(:user) { build_stubbed(:user, :moderator) }
 
   describe 'GET edit' do
     before { get :edit, id: user.id }
 
-    let(:user) { create(:user) }
+    context 'when authorized' do
+      let(:authorized?) { true }
+      let(:user) { create(:user, :admin) }
+
+      it_behaves_like 'a successful request'
+      it_behaves_like 'an edit request'
+    end
+
+    context 'when not authorized' do
+      let(:authorized?) { raise_pundit_error UserPolicy }
 
-    it_behaves_like 'a successful request'
-    it_behaves_like 'an edit request'
+      it_behaves_like 'an unauthorized access to a resource'
+    end
   end
 
   describe 'GET index' do
-    before do
-      get :index
+    before { get :index }
+
+    context 'when authorized' do
+      let(:authorized?) { true }
+      let(:user) { create(:user, :admin) }
+
+      it_behaves_like 'a successful request'
+      it_behaves_like 'an index request'
     end
 
-    it_behaves_like 'a successful request'
-    it_behaves_like 'an index request'
+    context 'when not authorized' do
+      let(:authorized?) { raise_pundit_error UserPolicy }
+
+      it_behaves_like 'an unauthorized access to a resource'
+    end
   end
 
   describe 'PATCH update' do
     before do
-      allow(UpdateUser).to receive(:call) { context }
-
       patch :update,
             id: user.id,
-            entry: {
-              name: Faker::Name.name,
-              is_active: [0, 1].sample
-            }
+            user: user_params
     end
 
-    let(:user) { create(:user) }
-    let(:context) { double(user: user, message: '', success?: success?) }
-    let(:success?) { fail 'success? not set' }
+    context 'when authorized' do
+      let(:authorized?) { true }
+      let(:user) { create(:user, :admin) }
+
+      context 'when the call is successful' do
```

Comment:
@LucasAnderson1984 I would recommend that you don't nest contexts, this is a smell that you're testing too much... its as bad as using an and in a describe :smile: